gnovm/pkg/gnolang/machine.go

```diff
@@ -371,11 +371,15 @@ func (m *Machine) Stacktrace() (stacktrace Stacktrace) {
 	for i := len(m.Frames) - 1; i >= 0; i-- {
 		fr := &m.Frames[i]
 		if fr.IsCall() && fr.Func.Name != "panic" {
-			calls = append(calls, StacktraceCall{
+			call := StacktraceCall{
 				CallExpr: fr.Source.(*CallExpr),
 				IsDefer:  fr.IsDefer,
-				FuncLoc:  fr.Func.GetSource(m.Store).GetLocation(),
-			})
+			}
+			fnSource := fr.Func.GetSource(m.Store)
+			if fnSource != nil {
+				call.FuncLoc = fnSource.GetLocation()
+			}
+			calls = append(calls, call)
```

Comment:
I think the issue you're trying to solve is solved here: https://github.com/gnolang/gno/pull/4192/files#r2052132917
Better to keep the machine simple and tailor the tests to suit it.

Reply:
rollbacked d34f5cf04